conda/resolve.py

```diff
@@ -764,22 +759,32 @@ def bad_installed(self, installed, new_specs):
         except TypeError:
             log.debug('Package set caused an unexpected error, assuming a conflict')
             solution = None
-        limit = None
+        limit = xtra = None
         if not solution or xtra:
             def get_(name, snames):
                 if name not in snames:
                     snames.add(name)
                     for fn in self.groups.get(name, []):
                         for ms in self.ms_depends(fn):
                             get_(ms.name, snames)
+            # New addition: find the largest set of installed packages that
+            # are consistent with each other, and include those in the
+            # list of packages to maintain consistency with
             snames = set()
+            try:
+                eq_optional_c = r2.generate_removal_count(C, specs)
+                solution, _ = C.minimize(eq_optional_c, C.sat())
+                snames.update(dists[Dist(q)]['name']
+                              for q in (C.from_index(s) for s in solution)
+                              if q and q[0] != '!' and '@' not in q)
+            except:
+                pass
```

Comment:
OK, I studied the history of [this line](https://github.com/conda/conda/blob/master/conda/resolve.py#L765) and it looks like we're clear to remove _both_ of the try/except blocks here. We put that in there [in this PR](https://github.com/conda/conda/pull/2251) 11 months ago, and I think we finally nailed the culprit [in this issue](https://github.com/conda/conda/issues/2276). We never removed the exception handling at that point, and I had since forgotten why it was there.